app/models/manageiq/providers/hawkular/datawarehouse_manager.rb

```diff
@@ -46,10 +46,17 @@ def self.raw_connect(hostname, port, token, alerts = false)
     end
 
     def connect(options = {})
-      @client ||= self.class.raw_connect(hostname,
-                                         port,
-                                         authentication_token('default'),
-                                         options[:alerts])
+      @clients ||= {}
+      @clients[options[:alerts] ? :alerts : :metrics] ||= self.class.raw_connect(
```

Comment:
There are only two options so It makes more sense to use a boolean instead of a string with two options (openshift cluster only exposes alerts and metrics, string is a type of metric)

Raw connect and connect have their reasons to exists - interfaces calling connect use a context
and raw_connect requires intimate knowledge of provider parameters. If you take a look at other providers they all look exactly the same. container_provider_mixin does something a little different: it passes the options forward, but I see no point in passing some of the parameters as options and some positionals.